packages/core/src/signers/privatekey.ts

```diff
@@ -0,0 +1,324 @@
+      if (this.signMode === SigningMode.AMINO) {
+        this.aminoSigner = await Secp256k1Wallet.fromKey(
+          this._privateKey.key,
+          "desmos"
+        );
+        [this.currentAccount] = await this.aminoSigner.getAccounts();
+      } else {
+        this.directSigner = await DirectSecp256k1Wallet.fromKey(
+          this._privateKey.key,
+          "desmos"
+        );
+        [this.currentAccount] = await this.directSigner.getAccounts();
+      }
+      this.updateStatus(SignerStatus.Connected);
+    } catch (e) {
+      this.clearSessionFields();
+      this.updateStatus(SignerStatus.NotConnected);
+      throw e;
+    }
+  }
+
+  async disconnect(): Promise<void> {
+    this.assertConnected();
+
+    this.updateStatus(SignerStatus.Disconnecting);
+    this.clearSessionFields();
+    try {
+      await this.keyProvider.disconnect();
+      this.updateStatus(SignerStatus.NotConnected);
+    } catch (e) {
+      this.updateStatus(SignerStatus.NotConnected);
+      throw e;
+    }
+  }
+
+  async getAccounts(): Promise<readonly AccountData[]> {
+    this.assertConnected();
+
+    if (this.signMode === SigningMode.AMINO) {
+      return this.aminoSigner!.getAccounts();
+    }
+    return this.directSigner!.getAccounts();
+  }
+
+  async getCurrentAccount(): Promise<AccountData | undefined> {
+    this.assertConnected();
+
+    return this.currentAccount;
+  }
+
+  async signAmino(
+    signerAddress: string,
+    signDoc: StdSignDoc
+  ): Promise<AminoSignResponse> {
+    this.assertConnected();
+
+    if (this.aminoSigner === undefined) {
+      throw new Error("sign format not supported");
+    }
+
+    return this.aminoSigner.signAmino(signerAddress, signDoc);
+  }
+
+  async signDirect(
+    signerAddress: string,
+    signDoc: SignDoc
+  ): Promise<DirectSignResponse> {
+    this.assertConnected();
+
+    if (this.directSigner === undefined) {
+      throw new Error("sign format not supported");
+    }
+
+    return this.directSigner.signDirect(signerAddress, signDoc);
+  }
+
+  get signingMode(): SigningMode {
+    return this.signMode;
+  }
+
+  get privateKey(): PrivateKey {
+    if (this._privateKey === undefined) {
+      throw new Error(
+        "can't get private key, check that the signer is connected"
+      );
+    }
+
+    return this._privateKey;
+  }
+
+  private static keyProviderStatusToSignerStatus(
+    status: PrivateKeyProviderStatus
+  ): SignerStatus {
+    switch (status) {
+      case PrivateKeyProviderStatus.NotConnected:
+        return SignerStatus.NotConnected;
+      case PrivateKeyProviderStatus.Connecting:
+        return SignerStatus.Connecting;
+      case PrivateKeyProviderStatus.Connected:
+        return SignerStatus.Connected;
+      case PrivateKeyProviderStatus.Disconnecting:
+        return SignerStatus.Disconnecting;
+      default:
+        throw new Error(`unknown status PrivateKeyProviderStatus: ${status}`);
```

Comment:
I do prefer keeping the switch, I think is clear and easy to maintain in case of extensions.
Also having a map in this case waste memory while the switch doesn't and also is ~6x faster. 
See [here](https://medium.com/front-end-weekly/switch-case-if-else-or-a-lookup-map-a-study-case-de1c801d944) for a peformance comparasion.